test/suite/lib.rs

```diff
@@ -63,74 +63,91 @@ async fn assert_balance(
     Ok(())
 }
 
-/// Checks whether the ledger supports token transfers and handles
-/// default sub accounts correctly.
-pub async fn test_transfer(ledger_env: impl LedgerEnv + LedgerEnv) -> TestResult {
+async fn transfer_or_fail(
+    ledger_env: &impl LedgerEnv,
+    amount: Nat,
+    receiver: Principal,
+) -> Result<Nat, TransferError> {
+    transfer(ledger_env, Transfer::amount_to(amount.clone(), receiver))
+        .await
+        .with_context(|| format!("failed to transfer {} tokens to {}", amount, receiver))
+        .unwrap()
+}
+
+async fn setup_test_account(
+    ledger_env: &(impl LedgerEnv + LedgerTransaction + std::clone::Clone),
+    amount: Nat,
+) -> anyhow::Result<impl LedgerEnv + LedgerTransaction> {
+    let balance = balance_of(ledger_env, ledger_env.principal()).await?;
+    assert!(balance >= amount.clone() + transfer_fee(ledger_env).await?);
     let receiver_env = ledger_env.fork();
     let receiver = receiver_env.principal();
+    assert_balance(&receiver_env, receiver, 0).await?;
```

Comment:
While you are at it can you also annotate `assert_balance` with `#[track_caller]`?